addons/hw_drivers/controllers/driver.py

```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/python3
 import logging
 import time
+import inspect
+from abc import ABCMeta
+from collections import deque
```

Comment:
why?

Reply:
I'm used to use it when having to append and prepend to a list but I suppose list.insert[0] makes more sense here since there is no performance issue. I'll change it.